--- ruby-1.8.7.299/test/readline/test_readline.rb.~1~	2010-09-20 15:52:19.575772353 +0200
+++ ruby-1.8.7.299/test/readline/test_readline.rb	2010-09-20 15:52:38.108647434 +0200
@@ -8,6 +8,8 @@ if defined?(Readline) && !/EditLine/n.ma
 require "test/unit"
 require "tempfile"

+ENV['TERM'] = 'dumb'
+
 class TestReadline < Test::Unit::TestCase
   def test_readline
     stdin = Tempfile.new("test_readline_stdin")
